### PR TITLE
fix!: Updates `expiration` paramater in `presign` functions to `TimeInterval` type.

### PR DESCRIPTION
**Pull request by epau**
After the upgrade to CRT 0.5.0+, the type of expiration changed from Int64 to TimeInterval.
Therefore, services that define presign functions (such as S3) fail to build because of the type mistmatch.

IMO it makes sense to update the type exposed in the functions to TimeInterval since TimeInterval is a better way to describe a value that should be in seconds.

## Issue \#
#804 

## Conventional Commits

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by epau** [comment]
Will regenerate models after this is merged.

**Review by jbelkins — APPROVED** [review]

**Review by ganeshnj — APPROVED** [review]

File: codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/customization/presignable/PresignableUrlIntegration.kt [hunk 1]
```diff
@@ -88,14 +87,14 @@ class PresignableUrlIntegration(private val presignedOperations: Map<String, Set
...
         writer.addImport(AWSClientRuntimeTypes.Core.AWSClientConfiguration)
         writer.addImport(ClientRuntimeTypes.Http.SdkHttpRequest)
+        writer.addFoundationImport()
 
         val httpBindingResolver = protocolGenerator.getProtocolHttpBindingResolver(protocolGeneratorContext, protocolGenerator.defaultContentType)
 
         writer.openBlock("extension $inputType {", "}") {
             writer.openBlock(
-                "public func presignURL(config: \$N, expiration: \$N) async throws -> \$T {", "}",
+                "public func presignURL(config: \$N, expiration: Foundation.TimeInterval) async throws -> \$T {", "}",
```
Comment (ganeshnj) [inline codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/customization/presignable/PresignableUrlIntegration.kt:96]: Could you add Foundation.TimeInterval to `FoundationTypes`?

Similar to SwiftTypes.

File: codegen/smithy-aws-swift-codegen/src/test/kotlin/software/amazon/smithy/aws/swift/codegen/PresignerGeneratorTests.kt [hunk 2]
```diff
@@ -73,9 +74,10 @@ class PresignerGeneratorTests {
             """
             import AWSClientRuntime
             import ClientRuntime
+            import Foundation
```
Comment (ganeshnj) [inline codegen/smithy-aws-swift-codegen/src/test/kotlin/software/amazon/smithy/aws/swift/codegen/PresignerGeneratorTests.kt:77]: would this work on Linux?

We should do partial import `import struct Foundation.TimeInterval`

**Review by ganeshnj — CHANGES_REQUESTED** [review]

File: codegen/smithy-aws-swift-codegen/src/test/kotlin/software/amazon/smithy/aws/swift/codegen/PresignerGeneratorTests.kt (hunk 2 above)
Comment (epau) [reply codegen/smithy-aws-swift-codegen/src/test/kotlin/software/amazon/smithy/aws/swift/codegen/PresignerGeneratorTests.kt:77]: will verify and will do!

File: codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/customization/presignable/PresignableUrlIntegration.kt (hunk 1 above)
Comment (epau) [reply codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/customization/presignable/PresignableUrlIntegration.kt:96]: makes sense, will do!

**Review by ganeshnj — APPROVED** [review]

**Pull request merged**